File: client-scalaz/src/main/scala/org/openapitools/client/api/ConsumerInviteResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ConsumerInviteResponse._

case class ConsumerInviteResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
inviterId: Option[Long],
inviter: Option[AccountShortResponse],
connection: Option[ConnectionResponse],
inviteText: Option[String],
token: Option[String],
albumCoverUrl: Option[String],
albumTitle: Option[String],
appName: Option[String],
welcomeText: Option[String],
application: Option[ApplicationShortResponse],
retailerLocation: Option[RetailerLocationResponse],
offer: Option[OfferShortResponse],
album: Option[AlbumResponse],
returning: Option[String])

object ConsumerInviteResponse {
  import DateTimeCodecs._

  implicit val ConsumerInviteResponseCodecJson: CodecJson[ConsumerInviteResponse] = CodecJson.derive[ConsumerInviteResponse]
  implicit val ConsumerInviteResponseDecoder: EntityDecoder[ConsumerInviteResponse] = jsonOf[ConsumerInviteResponse]
  implicit val ConsumerInviteResponseEncoder: EntityEncoder[ConsumerInviteResponse] = jsonEncoderOf[ConsumerInviteResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/consumer_invite_response.clj
(ns sirqul-io-t-platform.specs.consumer-invite-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.account-short-response :refer :all]
            [sirqul-io-t-platform.specs.connection-response :refer :all]
            [sirqul-io-t-platform.specs.application-short-response :refer :all]
            [sirqul-io-t-platform.specs.retailer-location-response :refer :all]
            [sirqul-io-t-platform.specs.offer-short-response :refer :all]
            [sirqul-io-t-platform.specs.album-response :refer :all]
            )
  (:import (java.io File)))


(def consumer-invite-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :inviterId) int?
   (ds/opt :inviter) account-short-response-spec
   (ds/opt :connection) connection-response-spec
   (ds/opt :inviteText) string?
   (ds/opt :token) string?
   (ds/opt :albumCoverUrl) string?
   (ds/opt :albumTitle) string?
   (ds/opt :appName) string?
   (ds/opt :welcomeText) string?
   (ds/opt :application) application-short-response-spec
   (ds/opt :retailerLocation) retailer-location-response-spec
   (ds/opt :offer) offer-short-response-spec
   (ds/opt :album) album-response-spec
   (ds/opt :returning) string?
   })

(def consumer-invite-response-spec
  (ds/spec
    {:name ::consumer-invite-response
     :spec consumer-invite-response-data}))
